tests/model_selection/test_hyperband.py

```diff
@@ -0,0 +1,398 @@
+            # power of search.aggressiveness
+            # In this case, assert that more models are tried then the max_iter
+            assert len(model_ids) > max_iter
+        else:
+            # Otherwise, give some padding "almost as many estimators are tried
+            # as max_iter". 3 is a fudge number chosen to be the minimum; when
+            # max_iter=20, len(model_ids) == 17.
+            assert len(model_ids) + 3 >= max_iter
+        assert all("bracket" in id_ for id_ in model_ids)
+
+    _test_basic()
+
+
+@pytest.mark.parametrize("max_iter,aggressiveness", [(27, 3), (30, 4)])
+def test_hyperband_mirrors_paper(max_iter, aggressiveness):
+    @gen_cluster(client=True, timeout=5000)
+    def _test_mirrors_paper(c, s, a, b):
+        X, y = make_classification(n_samples=10, n_features=4, chunks=10)
+        model = ConstantFunction()
+        params = {"value": np.random.rand(max_iter)}
+        alg = HyperbandSearchCV(
+            model,
+            params,
+            max_iter=max_iter,
+            random_state=0,
+            aggressiveness=aggressiveness,
+        )
+        yield alg.fit(X, y)
+        metadata = alg.metadata()
+        paper_decisions = [b.pop("decisions") for b in metadata["brackets"].values()]
+        actual_decisions = [
+            b.pop("decisions") for b in alg.metadata_["brackets"].values()
+        ]
+        assert metadata == alg.metadata_
+        for paper_iter, actual_iter in zip(paper_decisions, actual_decisions):
+            assert set(paper_iter).issubset(set(actual_iter))
+        if aggressiveness == 3:
+            assert alg.best_score_ == params["value"].max()
+
+    _test_mirrors_paper()
+
+
+@gen_cluster(client=True, timeout=5000)
+def test_hyperband_patience(c, s, a, b):
+    X, y = make_classification(n_samples=10, n_features=4, chunks=10)
+    model = ConstantFunction()
+    params = {"value": scipy.stats.uniform(0, 1)}
+    max_iter = 27
+
+    alg = HyperbandSearchCV(
+        model, params, max_iter=max_iter, patience=True, random_state=0
+    )
+    yield alg.fit(X, y)
+
+    alg_patience = max_iter // alg.aggressiveness
+    actual_decisions = [b.pop("decisions") for b in alg.metadata_["brackets"].values()]
+    paper_decisions = [b.pop("decisions") for b in alg.metadata()["brackets"].values()]
+
+    for paper_iter, actual_iter in zip(paper_decisions, actual_decisions):
+        trimmed_paper_iter = {k for k in paper_iter if k <= alg_patience}
+        assert trimmed_paper_iter.issubset(set(actual_iter))
+        assert all(x <= alg_patience + 1 for x in actual_iter)
+
+    assert alg.metadata_["partial_fit_calls"] <= alg.metadata()["partial_fit_calls"]
+    assert alg.best_score_ >= 0.9
+
+    alg = HyperbandSearchCV(model, params, max_iter=max_iter, patience=1)
+    with pytest.warns(UserWarning, match="The goal of `patience`"):
+        yield alg.fit(X, y)
+
+
+@gen_cluster(client=True, timeout=5000)
+def test_integration(c, s, a, b):
```

Comment:
I'd implement a check in here to make sure that `cv_results_` and `history_` are in the same order (i.e., that the have the same params and final score. ref: the comment above).

This test might be able to be trimmed down depending on IncrementalSearchCV's tests.

Reply:
I implemented this check